### PR TITLE
[SPARK-26288][CORE]add initRegisteredExecutorsDB

### PR DESCRIPTION
**Pull request by weixiuli**
## What changes were proposed in this pull request?
As we all know that  spark on Yarn uses  DB to record RegisteredExecutors information which  can  be reloaded  and  used  again  when the ExternalShuffleService is restarted . 

The RegisteredExecutors information can't be recorded  both in the mode of spark's standalone and  spark on k8s , which will cause the  RegisteredExecutors information to be lost ,when the ExternalShuffleService is restarted.

To solve the problem above, a method  is proposed and is committed .

## How was this patch tested?

new unit tests

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Comment by weixiuli** [comment]
@HyukjinKwon OK ,thank you!

**Comment by gatorsmile** [comment]
cc @jiangxb1987

**Comment by dongjoon-hyun** [comment]
Hi, @weixiuli . You can use `[CORE]` instead of `[ExternalShuffleService]`.

File: core/src/test/scala/org/apache/spark/deploy/worker/WorkerSuite.scala [hunk 1]
```diff
@@ -19,20 +19,20 @@ package org.apache.spark.deploy.worker
 
 import java.util.concurrent.atomic.AtomicBoolean
 import java.util.function.Supplier
-
```
Comment (dongjoon-hyun) [inline core/src/test/scala/org/apache/spark/deploy/worker/WorkerSuite.scala]: Please execute `dev/scalastyle` to check the coding style.

File: core/src/test/scala/org/apache/spark/deploy/worker/WorkerSuite.scala [hunk 2]
```diff
@@ -243,4 +243,13 @@ class WorkerSuite extends SparkFunSuite with Matchers with BeforeAndAfter {
...
     assert(cleanupCalled.get() == value)
   }
+  test("test  initRegisteredExecutorsDB  ") {
+    val sparkConf = new SparkConf()
+    Utils.loadDefaultSparkProperties(sparkConf)
+    val securityManager = new SecurityManager(sparkConf)
+    sparkConf.set(config.SHUFFLE_SERVICE_DB_ENABLED.key, "true")
+    sparkConf.set(config.SHUFFLE_SERVICE_ENABLED.key, "true")
+    sparkConf.set("spark.local.dir", "/tmp")
+    val externalShuffleService = new ExternalShuffleService(sparkConf, securityManager)
```
Comment (dongjoon-hyun) [inline core/src/test/scala/org/apache/spark/deploy/worker/WorkerSuite.scala]: Does this test case fail without your patch?

Comment (weixiuli) [reply core/src/test/scala/org/apache/spark/deploy/worker/WorkerSuite.scala]: @dongjoon-hyun  Thanks a lot,I have fixed scalastyle  and removed  the test ablove  yet.
There is a new test to test this patch. 
Kindly review.

File: core/src/main/scala/org/apache/spark/internal/config/package.scala [hunk 3]
```diff
@@ -182,6 +182,12 @@ package object config {
   private[spark] val SHUFFLE_SERVICE_ENABLED =
     ConfigBuilder("spark.shuffle.service.enabled").booleanConf.createWithDefault(false)
 
+  private[spark] val SHUFFLE_SERVICE_DB_ENABLED =
+    ConfigBuilder("spark.shuffle.service.db.enabled")
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/internal/config/package.scala]: what's the reason this is off by default?

Comment (weixiuli) [reply core/src/main/scala/org/apache/spark/internal/config/package.scala]: Hi，@rxin，There is no reason for that，I also  think it's better left on by default and I will fix it.

**Comment by weixiuli** [comment]
retest please

**Pull request closed without merging**